flycheck.el

```diff
@@ -3610,6 +3650,55 @@ list."
       ;; highlighting
       (flycheck-error-list-highlight-errors preserve-pos))))
 
+(defvar flycheck-error-list-current-filter nil
+  "The minimum level of errors to display in the error list.
+
+If set to an error level, only displays errors whose error level
+is at least as severe as this one in the error list.  If nil,
+display all errors.")
+
+(defun flycheck-error-list-mode-line-filter-indicator ()
+  "Create a string representing the current error list filter."
+  (if flycheck-error-list-current-filter
+      (format " [>= %s]" (symbol-name flycheck-error-list-current-filter))
+    ""))
+
+(defun flycheck-error-list-query-level (prompt)
+  "Reads an error level from the user.
+
+Only offers level for which errors currently exist."
+  (-when-let* ((error-levels (mapcar #'flycheck-error-level
+                                     (flycheck-error-list-current-errors)))
+               (uniq-error-levels (-uniq error-levels))
```

Comment:
We do not track them, but we can find all levels the same way we find all syntax checkers at some places: By iterating over all symbols and testing for special properties.

Assuming that we'll at some point have dozen of packages with custom error types, that's not a good idea, but I don't think that we ever will.  Actually, I'm not aware of any extension that actually defines custom levels, and in fact I only know of a single external user of `flycheck-define-error-level`: Spacemacs, which uses it to redefine the built-in levels to change their fringe indicators.

But then again, under this assumption we don't need to spend much care for custom levels either, so I think that your approach of including the built-in levels and the currently present levels, and allowing the user to input arbitrary symbols is a good compromise.

Let's keep it that way.  If it doesn't work well for our users, we can still fix it afterwards.